charms/istio-pilot/src/charm.py

```diff
@@ -293,23 +293,38 @@ def handle_default_gateway(self, event):
         # Update the ingress resources as they rely on the default_gateway
         self.handle_ingress(event)
 
-    def handle_gateway_info_relation(self, event):
-        if not self.model.relations["gateway-info"]:
-            self.log.info("No gateway-info relation found")
-            return
+    def handle_gateway_info_relation(self, _) -> None:
+        """Sends the default-gateway info through the gateway-info relation.
+
+        Raises:
+            GenericCharmRuntimeError("default-gateway not present in cluster):
+                if the default-gateway is not present in the cluster
+        """
+        # Verify the default-gateway is present in the cluster
         is_gateway_created = self._resource_handler.validate_resource_exist(
             resource_type=self._resource_handler.get_custom_resource_class_from_filename(
                 "gateway.yaml.j2"
             ),
             resource_name=self.model.config["default-gateway"],
             resource_namespace=self.model.name,
         )
-        if is_gateway_created:
-            self.gateway.send_gateway_relation_data(
-                self.app, self.model.config["default-gateway"], self.model.name
-            )
-        else:
+
+        # Raise if the default-gateway is not created, it must be present always
+        if not is_gateway_created:
             self.log.info("Gateway is not created yet. Skip sending gateway relation data.")
+            raise GenericCharmRuntimeError("default-gateway not present in cluster.")
```

Comment:
I do not think we need to raise the error here. That will put the unit into error state

Reply:
I was thinking this a lot, but to me if there is no default-gateway, istio-pilot is not doing its job correctly. I was not sure to raise this exception here because this should be handled elsewhere in the charm code, but as a check is better.

Reply:
I think this gets refactored by the stuff I'm doing right now anyway, so not important to battle it out here imo.  We can debate it in a day on mine if needed :)

Reply:
need to understand whose code will land faster. I also do not want to block Daniela

Reply:
Well, this change won't block any kserve work. I only needed the library, this PR is a consequence of that change and only applies to istio-pilot. We can have minimal changes in this PR and let @ca-scribner do the rest.

Reply:
yeah I say we resolve this thread and it'll get sorted out by end of week anyway